src/local_neighborhood.py

```diff
@@ -0,0 +1,121 @@
+        @param network:  input network file (required)
+        @param output_file: path to the output pathway file (required)
+        @param k: path length (optional)
+        @param singularity: if True, run using the Singularity container instead of the Docker container
+        """
+        # Add additional parameter validation
+        # Do not require k
+        # Use the PathLinker default
+        # Could consider setting the default here instead
+        if not nodetypes or not network or not output_file:
+            raise ValueError('Required PathLinker arguments are missing')
+
+        work_dir = '/spras'
+
+        # Each volume is a tuple (src, dest)
+        volumes = list()
+
+        bind_path, node_file = prepare_volume(nodetypes, work_dir)
+        print("bind_path: ", bind_path)
+        print("node_file: ", node_file)
+
+        volumes.append(bind_path)
+        print("volumes1: ", volumes)
+
+        bind_path, network_file = prepare_volume(network, work_dir)
+        volumes.append(bind_path)
+        print("volumes2: ", volumes)
+
+        # PathLinker does not provide an argument to set the output directory
+        # Use its --output argument to set the output file prefix to specify an absolute path and prefix
+        out_dir = Path(output_file).parent
+        print("out_dir: ", out_dir)
+        # PathLinker requires that the output directory exist
+        out_dir.mkdir(parents=True, exist_ok=True)
+        bind_path, mapped_out_dir = prepare_volume(str(out_dir), work_dir)
+        volumes.append(bind_path)
+        print("bind_path: ", bind_path)
+        print("mapped_out_dir: ", mapped_out_dir)
+        print("bind_path: ", bind_path)
+        mapped_out_prefix = mapped_out_dir + '/out'  # Use posix path inside the container
+
+        command = ['python',
+                   '/PathLinker/run.py',
+                   network_file,
+                   node_file,
+                   '--output', mapped_out_prefix]
+
+        # Add optional argument
+        if k is not None:
+            command.extend(['-k', str(k)])
+
+        print('Running PathLinker with arguments: {}'.format(' '.join(command)), flush=True)
+
+        # TODO consider making this a string in the config file instead of a Boolean
+        container_framework = 'singularity' if singularity else 'docker'
+        out = run_container(container_framework,
+                            'reedcompbio/pathlinker',
+                            command,
+                            volumes,
+                            work_dir)
+        print(out)
+
+        # Rename the primary output file to match the desired output filename
+        # Currently PathLinker only writes one output file so we do not need to delete others
+        # We may not know the value of k that was used
+        output_edges = Path(next(out_dir.glob('out*-ranked-edges.txt')))
+        output_edges.rename(output_file)
```

Comment:
You shouldn't need to rename the output file if the volume mapping worked correctly.